solutions/usaco-838.mdx

```diff
@@ -2,11 +2,15 @@
 id: usaco-838
 source: USACO Gold 2018 Open
 title: Milking Order
-author: Qi Wang
+author: Timothy Gao and Qi Wang
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_milkorder_gold_open18.html)
 
+Let's draw a directed edge from each node $u$ to another node $v$, if we node that $u$ must come before $v$. Note that this can be done when we process the $M$ observations. Now, notice that an ordering is valid only if the resulting graph is a DAG.  Additionally, we can use a PriorityQueue(instead of a queue) with topological sort to generate the lexographically minimum topological ordering. It remains to find how many observations we can follow starting from the first observation, which we can binary search for.
```

Comment:
```suggestion
Let's draw a directed edge from each node $u$ to another node $v$, if we node that $u$ must come before $v$. Note that this can be done when we process the $M$ observations. Now, notice that an ordering is valid only if the resulting graph is a DAG. Additionally, we can use a PriorityQueue(instead of a queue) with topological sort to generate the lexicographically minimum topological ordering. It remains to find how many observations we can follow starting from the first observation, which we can binary search for.
```